server/src/main/java/org/apache/druid/server/QueryResource.java

```diff
@@ -364,7 +341,12 @@ public void write(OutputStream outputStream) throws WebApplicationException
 
       log.noStackTrace()
          .makeAlert(e, "Exception handling request")
-         .addData("query", query != null ? jsonMapper.writeValueAsString(query) : "unparseable query")
```

Comment:
Nit: The original version seemed more concise. As we are using `queryLifecycle.getQuery()` in several places, assigning to a final variable seems reasonable.

Reply:
This is an intentional change. It doesn't seem like a good convention to use an object after the object is moved to inside of another. Note that `Query` is immutable and you need to create a new copy when you annotate it with extra information. Even though this is not currently happening in `QueryLifecycle`, it might be able to happen later in the future. Then, the `query` in `QueryResource` will not be guaranteed to be the same object as the `query` in `QueryLifecycle`.

Reply:
Yeah, makes sense.